bpo-29275: Remove Y2K reference from time module docs

calluw opened the pull request:
The Y2K reference is no longer needed as it only points out that Python's use of C standard functions doesn't generally suffer from Y2K issues; the point regarding conventions for conversion of 2-digit years in :func:`strptime` is still valid.

Some advice on what indexes to add/remove would be appreciated alongside review.

<!-- issue-number: [bpo-29275](https://bugs.python.org/issue29275) -->
https://bugs.python.org/issue29275
<!-- /issue-number -->

Commit:
Remove the Y2K reference from time doc warning about 2-digit year conversion

The Y2K reference is not needed as it only points out that Python's use
of C standard functions doesn't generally suffer from Y2K issues; the
point regarding conventions for conversion of 2-digit years in
:func:`strptime` is still valid.

benjaminp commented on Doc/library/time.rst:
```diff
@@ -43,16 +43,13 @@ An explanation of some terminology and conventions is in order.
 .. index::
    single: Year 2000
+   single: 2-digit years
    single: Y2K
```
I think you should remove these old index entries, too.

calluw replied:
I wasn't sure if they'd still be useful as redirection, but agree they're likely not needed and will remove.

calluw committed:
Remove additional index references to Y2K/Year 2000

These are likely no longer needed as the paragraph now no longer refers to them directly as topics.

pganssle reviewed (commented):
Bit late to the game, but thanks for this PR!

Hopefully this doesn't result in some weird box-checking certification process deciding that we're no longer Y2K compliant 😄.
